src/Misc/virtualJoystick.ts

```diff
@@ -284,6 +378,70 @@ export class VirtualJoystick {
         this._joystickColor = newColor;
     }
 
+    /**
+    * Sets the size of the puck
+    * @param newSize puck size
+    */
+    public setPuckSize(newSize: number) {
+        this._joystickPuckSize = newSize;
+        this._clearPuckSize = ~~(this._joystickPuckSize * 2.1);
+        this._clearPuckSizeOffset = ~~(this._clearPuckSize / 2);
+    }
+
+    /**
+    * Sets the size of the container
+    * @param newSize container size
+    */
+    public setContainerSize(newSize: number) {
+        this._joystickContainerSize = newSize;
+        this._clearContainerSize = ~~(this._joystickContainerSize * 2.1);
+        this._clearContainerSizeOffset = ~~(this._clearContainerSize / 2);
+    }
+
+    /**
+    * Clears the set position of the joystick
+    */
+    public clearPosition() {
+        this._alwaysVisible = false;
+
+        delete this._joystickPosition;
+    }
+
+    /**
+    * Sets whether or not joystick container is always visible.
+    * Can only be always visible if `_joystickPosition` is set
+    * @param alwaysVisible defines whether or not the joystick container is always visible
+    */
+    public setAlwaysVisible(alwaysVisible: boolean) {

```

Comment:
Maybe a property?

Reply:
@deltakosh 

I made a method for this since it didn't make sense to me to have a joystick always visible if it doesn't have a set position, so modifying this behavior via a method allows for a check that position is set.

Do you think `alwaysVisible` should be allowed whether or not a fixed position is set?